implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -0,0 +1,261 @@
+
+    private static final ThreadFactory DEFAULT_THREAD_FACTORY = new NamedThreadFactory("otlp-metrics-publisher");
+    private final OtlpConfig config;
+    private final HttpSender httpSender;
+
+    private final Resource resource;
+
+    public OtlpMeterRegistry() {
+        this(OtlpConfig.DEFAULT, Clock.SYSTEM);
+    }
+
+    public OtlpMeterRegistry(OtlpConfig config, Clock clock) {
+        this(config, clock, new HttpUrlConnectionSender());
+    }
+
+    public OtlpMeterRegistry(OtlpConfig config, Clock clock, HttpSender httpSender) {
+        super(config, clock);
+        this.config = config;
+        this.httpSender = httpSender;
+        this.resource = Resource.newBuilder()
+                .addAllAttributes(getResourceAttributes())
+                .build();
+        start(DEFAULT_THREAD_FACTORY);
+    }
+
+    @Override
+    protected void publish() {
+        for (List<Meter> batch : MeterPartition.partition(this, config.batchSize())) {
+            List<Metric> metrics = batch.stream()
+                    .map(meter -> meter.match(
+                            this::writeGauge,
+                            this::writeCounter,
+                            this::writeHistogramSupport,
+                            this::writeHistogramSupport,
+                            this::writeHistogramSupport,
+                            this::writeGauge,
+                            this::writeMeter,
+                            this::writeMeter,
+                            this::writeMeter))
+                    .collect(Collectors.toList());
+
+            try {
+                ExportMetricsServiceRequest request = ExportMetricsServiceRequest.newBuilder()
+                        .addResourceMetrics(ResourceMetrics.newBuilder()
+                                .setResource(this.resource)
+                                .addScopeMetrics(ScopeMetrics.newBuilder()
+                                        // we don't have instrumentation library/version attached to meters; leave unknown for now
+                                        //.setScope(InstrumentationScope.newBuilder().setName("").setVersion("").build())
+                                        .addAllMetrics(metrics)
+                                        .build())
+                                .build())
+                        .build();
+                this.httpSender.post(this.config.url())
+                        .withContent("application/x-protobuf", request.toByteArray())
+                        .send();
+            } catch (Throwable e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    protected <T> Gauge newGauge(Meter.Id id, T obj, ToDoubleFunction<T> valueFunction) {
+        return new DefaultGauge<>(id, obj, valueFunction);
+    }
+
+    @Override
+    protected Counter newCounter(Meter.Id id) {
+        return new StartTimeAwareCumulativeCounter(id, this.clock);
```

Comment:
The folks who own my metrics backend consistently ask me if we can support delta counters as part of our migration to OTLP. Do we have any thoughts yet on an implementation for delta counters in Micrometer?

Reply:
It's configurable in `SimpleMeterRegistry`. We could make it configurable in this registry too, since OTLP the format doesn't specify a preference as far as I'm aware.
What's the motivation behind the preference for delta counts?

Reply:
That's a bit beyond my expertise, but supposedly it will allow for more accurate results in our use case and improve the query performance for at least one of our metrics backends. The Wavefront folks did a short writeup on pros/cons in their docs -> https://docs.wavefront.com/delta_counters.html

Using the same config options as SimpleMeterRegistry seems reasonable, but you also need to carry the context forward and set the right `AggregationTemporality`

Reply:
I've added it as a follow-up to make the temporality configurable. We'll see about doing it before 1.9 GA if it's important to users. I've left it out of this pull request for the sake of simplicity and time to get RC1 out.